File: apps/dotnet/Simex/Simex/Models/OperationStateHistory.cs
using System;
using System.Collections.Generic;

namespace Simex.Models;

public partial class OperationStateHistory
{
    public int Id { get; set; }

    public DateTime Date { get; set; }

    public int? OperationStateId { get; set; }

    public int? OperationId { get; set; }

    public string? Observations { get; set; }

    public virtual Operation? Operation { get; set; }

    public virtual OperationState? OperationState { get; set; }
}

File: apps/web/backend/app/Models/OperationStateHistory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class OperationStateHistory extends Model
{
    protected $table = 'OPERATION_STATE_HISTORY';

    protected $primaryKey = 'ID';

    public $timestamps = false;

    protected $fillable = [
        'ID',
        'DATE',
        'OPERATION_STATE_ID',
        'OPERATION_ID',
        'OBSERVATIONS',
    ];

    public function Operation()
    {
        return $this->belongsTo(Operations::class, 'OPERATION_ID');
    }

    public function OperationState()
    {
        return $this->belongsTo(OperationStates::class, 'OPERATION_STATE_ID');
    }
}
